app/src/main/java/com/example/bigowlapp/SearchContactsToSupervise.java

```diff
@@ -0,0 +1,130 @@
+package com.example.bigowlapp;
+
+import android.content.ContentResolver;
+import android.database.Cursor;
+import android.os.Bundle;
+import android.provider.ContactsContract;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.util.Log;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ListView;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.example.bigowlapp.model.User;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SearchContactsToSupervise extends AppCompatActivity {
+   // private TextView listContacts;
+    private ListView listContactsView;
+    private List<String> list, listShow;
+    private Button loadContacts;
+    private FirebaseFirestore db = FirebaseFirestore.getInstance();
+    private static ArrayList<QueryDocumentSnapshot> qds = new ArrayList<QueryDocumentSnapshot>();
+
+    EditText search_users;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_search_contacts);
+        initialize();
+    }
+
+    protected void initialize() {
+        //listContacts = (TextView) findViewById(R.id.listContacts);
+        /*loadContacts = (Button) findViewById(R.id.loadContacts);
+        loadContacts.setOnClickListener(new View.OnClickListener(){
+            @Override
+            public void onClick(View view) {*/
+                loadContacts();
+            //}
+        //});
+
+    }
+
+    private void loadContacts() {
+        //StringBuilder builder = new StringBuilder();
+        ContentResolver contentResolver = getContentResolver();
+        Cursor cursor = contentResolver.query(ContactsContract.Contacts.CONTENT_URI, null,null,null,null);
```

Comment:
Got a runtime exceotion "requires android.permission.READ_CONTACTS or android.permission.WRITE_CONTACTS". Pop up the system alert window for the app to gain permission from user.

Reply:
fixed

Reply:
fixed